### DISCUSSION TITLE
Timing issue on multi dependency updates

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#24054] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Describe the problem

In [my project](https://github.com/nielsbasjes/yauaa) I have dependencies on external libraries.
In some cases this dependency is really the dependency on multiple jar files that all need to be the same version for everything to work.

In this example it is Apache Nifi; I have seen the same problem with other multi-jar dependency projects also.

In my pom.xml I have [a single property](https://github.com/nielsbasjes/yauaa/blob/bc8e04273a570c4628a4c1284ff6cbc7a7285c42/pom.xml#L74) with the desired version that is then used in [multiple dependencies](https://github.com/nielsbasjes/yauaa/blob/bc8e04273a570c4628a4c1284ff6cbc7a7285c42/udfs/nifi/nifi-processors/pom.xml#L44-L61) in my project.

The intermittent problem I have seen is:

1) Such a project releases a new version. 
2) A release of many dependencies is not atomic; the various parts appear one by one in maven central which takes "a while".
3) While the release of this project is still on going: Renovate at my end sees for (at least) one of those dependencies that a new version has arrived and provides a patch (only updating the above shown property).

So the pull request shows
![image](https://github.com/renovatebot/renovate/assets/240096/61e7c50e-2537-4589-8d7b-febcc5a2a2ea)

with the only code change being 
![image](https://github.com/renovatebot/renovate/assets/240096/8cf969a4-c43b-479a-a280-44b346a799aa)

4) Github Actions fires on the branch with this patch (note that the release has not yet finished) and fails with errors about missing dependencies:

Copied from https://github.com/nielsbasjes/yauaa/actions/runs/5944210412/job/16120993668

```
Error:  Failed to execute goal on project yauaa-nifi-parent: 
Could not resolve dependencies for project nl.basjes.parse.useragent:yauaa-nifi-parent:pom:7.22.1-SNAPSHOT: 
The following artifacts could not be resolved: 
org.apache.nifi:nifi-framework-api:jar:1.23.2 (absent), 
org.apache.nifi:nifi-runtime:jar:1.23.2 (absent), 
org.apache.nifi:nifi-nar-utils:jar:1.23.2 (absent), 
org.apache.nifi:nifi-properties:jar:1.23.2 (absent): 
org.apache.nifi:nifi-framework-api:jar:1.23.2 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. 
This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

A short while later the release of all the dependency jars completes and then Renovate updates the pull request to show this 

![image](https://github.com/renovatebot/renovate/assets/240096/b1875c18-1a80-4b69-a7b6-8076b6341129)

I suspect that because the code does not change the Github Actions does not fire automatically.

Now when I rerun the same job manually it (usually) passes successfully.

Possible solution directions I see right now:
1. Change nothing; So it first fails and then automatically passes a while later when you manually rerun it.
2. Change nothing; Only re-run the Github Actions if such a change occurs. So it first fails and then automatically passes a while later.
3. Only create a merge request if all applicable dependencies exist. Not a good idea because projects tend to change the applicable dependencies with major versions and I want to be automatically informed that I need to fix something.

Right now I would vote on "2." to automatically rerun the build.

I do not know how to provide a reproduction for this.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment, marked as answer]
A workaround for this would be to set a https://docs.renovatebot.com/configuration-options/#minimumreleaseage setting long enough that each package is nearly always updated by then.

Alternatively there have been discussions in the past about a new feature where you can configure something like a "minimum updates" value to ensure they're all updated, but that could be painful to maintain manually

**Reply by nielsbasjes** [reply]
I think setting the minimumReleaseAge to 1 day is good enough for my projects.
Thanks!

**Reply by HonkingGoose** [reply]
> Alternatively there have been discussions in the past about a new feature where you can configure something like a "minimum updates" value to ensure they're all updated, but that could be painful to maintain manually

Relevant issues:

- #12914
- #21368

**Reply by nielsbasjes** [reply]
Just to double check;
Can Renovate trigger a rerun of the Github Actions (or Gitlab CI) when such an update of the pull request occurs which does NOT have any additional code changes? 
Can Renovate (be modified to) fire an API call that does this?
Or can these CI systems only look at the code changes to re-run?

**Reply by rarkins** [reply]
The code platforms themselves decide what triggers CI. Typically it's any time a commit is pushed
